compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -13357,11 +13359,15 @@ private STNode parseAnonFuncExprOrFuncTypeDesc(List<STNode> qualifiers) {
                 return parseExpressionRhs(DEFAULT_OP_PRECEDENCE, anonFunc, false, true);
             case IDENTIFIER_TOKEN:
             default:
-                switchContext(ParserRuleContext.VAR_DECL_STMT);
                 STNode funcTypeDesc = STNodeFactory.createFunctionTypeDescriptorNode(qualifierList, functionKeyword,
                         funcSignature);
-                return parseComplexTypeDescriptor(funcTypeDesc, ParserRuleContext.TYPE_DESC_IN_TYPE_BINDING_PATTERN,
-                        true);
+                if (!isInBracketedList) {
```

Comment:
Rather than the flag, can we use the ParserRuleContext to determine where we are?

Reply:
Context information is not available and it's not straightforward to pass the ctx from calling methods. I think using a flag is better here. WDYT?

Reply:
Catch is, a bracketed list can also exist in a nested level as well. e.g: ` [ [function( ], x]`. 
But Since you are passing hardcoded `false` in L#13248 and L#13324, this condition may sometimes return false for valid scenarios.

If passing down the context is a pain, We can add a method to get the current context, similar to add/end/switch context methods.

Reply:
Done